Dockerfile

```diff
@@ -1,23 +1,36 @@
-FROM ubuntu:22.04 AS build
+# Since we support arm64, take advantage of Docker and Zig's cross
+# compilation, rather than running under qemu.
+# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
+FROM --platform=$BUILDPLATFORM ubuntu:22.04 AS build
 
 RUN  apt-get update \
   && apt-get install -y wget xz-utils
 
 WORKDIR /opt/beta-beetle
 
-COPY src ./src
-COPY scripts ./scripts
-COPY build.zig ./build.zig
+# Copy install_zig by itself, to try and keep cache hits as high as possible.
+RUN mkdir scripts
+COPY scripts/install_zig.sh ./scripts
 
 ENV PATH="${PATH}:/opt/beta-beetle/zig"
 
 # Allows passing in --build-arg DEBUG=true during builds (needed by Github CI for generating debug image)
 ARG DEBUG=false
+
+# With our multiplatform builds, Docker will only repeat steps after the ARG TARGETPLATFORM.
+# So, explicitly install zig before then, to not need to do it twice.
+RUN ./scripts/install_zig.sh
+
+COPY scripts ./scripts
+COPY src ./src
+COPY build.zig ./build.zig
+
+ARG TARGETPLATFORM
 RUN ./scripts/install.sh
 
-FROM ubuntu:22.04 AS release
-WORKDIR /opt/beta-beetle
+FROM scratch AS release
```

Comment:
Maybe we can do alpine or busybox instead of scratch so people still have access to `ls`, `ps`, `netstat`, whatnot.